

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>7. DO-IT-YOURSELF &mdash; Physalia Paleogenomics 0.1.0 documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="8. Material from previous courses" href="10_old_material.html" />
    <link rel="prev" title="6. Filtering, annotating and combining SNPs" href="8_Filtering_SNPs.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home" alt="Documentation Home"> Physalia Paleogenomics
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="1_ListTools.html">1. List of Tools</a></li>
<li class="toctree-l1"><a class="reference internal" href="2_ReadsFiltering.html">2. Quality filtering of reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="3_Metagenomics_v2.html">3. Metagenomic screening of shotgun data</a></li>
<li class="toctree-l1"><a class="reference internal" href="4_ReadsMapping_v2.html">4. Alignment of reads to a reference genome</a></li>
<li class="toctree-l1"><a class="reference internal" href="7_VariantsCall.html">5. Variant calling and visualization</a></li>
<li class="toctree-l1"><a class="reference internal" href="8_Filtering_SNPs.html">6. Filtering, annotating and combining SNPs</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">7. DO-IT-YOURSELF</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#hands-on-1-ancient-human-mtdna">7.1. Hands-on 1: ancient human mtDNA</a></li>
<li class="toctree-l2"><a class="reference internal" href="#hands-on-2-ancient-human-dental-calculus">7.2. Hands-on 2: ancient human dental calculus</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#r-session">7.2.1. R session</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#upgma">7.2.1.1. UPGMA</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="10_old_material.html">8. Material from previous courses</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Physalia Paleogenomics</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li><span class="section-number">7. </span>DO-IT-YOURSELF</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/9_DIY.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="do-it-yourself">
<h1><span class="section-number">7. </span>DO-IT-YOURSELF<a class="headerlink" href="#do-it-yourself" title="Permalink to this headline">¶</a></h1>
<div class="section" id="hands-on-1-ancient-human-mtdna">
<h2><span class="section-number">7.1. </span>Hands-on 1: ancient human mtDNA<a class="headerlink" href="#hands-on-1-ancient-human-mtdna" title="Permalink to this headline">¶</a></h2>
<p>In this final hands-on session you will analyse shotgun (reduced) sequencing data generated from an ancient human tooth.The genomic library built from the DNA extract was sequenced on an Illumina platform in paired-end mode. Your task is:</p>
<ol class="arabic simple">
<li><p>Process the raw reads (remove adapters, merge the reads, section 2).</p></li>
<li><p>Align the reads to the human mitochondrial DNA (mtDNA) reference sequence, assess the damage of DNA molecules, call the variants (sections 4-5-6).</p></li>
<li><p>Run the metagenomic screning of the DNA extract with Kraken using the Minikraken database (section 3).</p></li>
</ol>
<p>After reads pre-processing it is up to you whether first aliging the reads or screening the metagenomic content.</p>
<ul>
<li><p><strong>Option 1</strong>: You can use your <code class="docutils literal notranslate"><span class="pre">vcf</span></code> file to assign an haplogroup to the human samples that you analysed. Some useful tools for haplogroup assignation:</p>
<blockquote>
<div><ul class="simple">
<li><p>Check the variant positions in Phylotree (<a class="reference external" href="http://www.phylotree.org/">http://www.phylotree.org/</a>)</p></li>
<li><p>Load the <code class="docutils literal notranslate"><span class="pre">vcf</span></code> file in Haplogrep (<a class="reference external" href="https://haplogrep.uibk.ac.at">https://haplogrep.uibk.ac.at</a>)</p></li>
</ul>
</div></blockquote>
</li>
<li><p><strong>Option 2</strong>: Run again the metagenomic screening with a Custom Database of Kraken (provided by us), and compare the results with those obtained with Minikraken.</p></li>
</ul>
</div>
<div class="section" id="hands-on-2-ancient-human-dental-calculus">
<h2><span class="section-number">7.2. </span>Hands-on 2: ancient human dental calculus<a class="headerlink" href="#hands-on-2-ancient-human-dental-calculus" title="Permalink to this headline">¶</a></h2>
<p>In this hands-on you will use a sequence dataset of ancient dental calculus from a recent study by <a class="reference external" href="https://link.springer.com/article/10.1186/s40168-019-0717-3">Velsko et al.</a> (Microbiome, 2019), which consists of paired-end sequence data.
After filtering and collapsing the reads, classify the reads of each sample with Kraken2 and estimate the abudances with Bracken.</p>
<blockquote>
<div></div></blockquote>
<p>After that, you can use R to run analyses and create charts from the abundance tables generated with Bracken.
Run the following R script to merge the species aundances of all the samples in one table. The script needs as argument the path to the folder containing the bracken results.
Move to the bracken results folder and type the following command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">Rscript</span> <span class="n">brackenToAbundanceTable</span><span class="o">.</span><span class="n">R</span> <span class="o">.</span>
</pre></div>
</div>
<p>The script will generate two abundance tables, <code class="docutils literal notranslate"><span class="pre">taxa_abundance_bracken_IDs.txt</span></code>, which contains the species names as NCBI IDs, and <code class="docutils literal notranslate"><span class="pre">taxa_abundance_bracken_names.txt</span></code>, which contains the actual species names.
Since the Minikraken database was built from complete Bacterial, Archaeal and Viral genomes, we must make a normalization for the genome leghts of each species. This normalization is important when we want to analyse relative taxa abundances to characterise full microbiomes.
To do that will use a Python script that takes three arguments:</p>
<ol class="arabic simple">
<li><p>The abundance table</p></li>
<li><p>The table of genome legths</p></li>
<li><p>The name of the output file</p></li>
</ol>
<p>To run the normalization type the command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">python</span> <span class="n">gL</span><span class="o">-</span><span class="n">normalizer</span><span class="o">-</span><span class="n">lite</span><span class="o">.</span><span class="n">py</span> <span class="n">taxa_abundance_bracken_names</span><span class="o">.</span><span class="n">txt</span> <span class="n">prokaryotes_viruses_organelles</span><span class="o">.</span><span class="n">table</span> <span class="n">taxa_abundance_bracken_names_normalized</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
<div class="section" id="r-session">
<h3><span class="section-number">7.2.1. </span>R session<a class="headerlink" href="#r-session" title="Permalink to this headline">¶</a></h3>
<p>Now can download the normalized table in your local machine (e.g. with ´´scp´´) and open R.</p>
<p>First of all, in R, we must set up the folder which contains the abundance table:</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">setwd</span><span class="p">(</span><span class="s">&quot;/path/to/your/folder&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p>Then we can import the abundance table files, setting the 1st column as row names. Then we remove the 1st, 2nd and 3rd column, which will not be used for the analysis.</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">table.species</span> <span class="o">=</span> <span class="nf">read.delim</span><span class="p">(</span><span class="s">&quot;taxa_abundance_bracken_names_normalized.txt&quot;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="bp">T</span><span class="p">,</span> <span class="n">fill</span><span class="o">=</span><span class="bp">T</span><span class="p">,</span> <span class="n">row.names</span><span class="o">=</span><span class="m">1</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">)</span>
<span class="n">table.species.final</span> <span class="o">=</span> <span class="n">table.species</span><span class="p">[,</span><span class="o">-</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">3</span><span class="p">)]</span>
</pre></div>
</div>
<p>Then we filter the species in the table for their abundance by removing those that are represented below a threshold of 0.02%.
First we define a function (that we call low.count.removal)</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">low.count.removal</span> <span class="o">=</span> <span class="nf">function</span><span class="p">(</span>
                        <span class="n">data</span><span class="p">,</span>                 <span class="c1"># OTU count data frame of size n (sample) x p (OTU)</span>
                        <span class="n">percent</span><span class="o">=</span><span class="m">0.02</span>  <span class="c1"># cutoff chosen</span>
                        <span class="p">){</span>
    <span class="n">keep.otu</span> <span class="o">=</span> <span class="nf">which</span><span class="p">(</span><span class="nf">colSums</span><span class="p">(</span><span class="n">data</span><span class="p">)</span><span class="o">*</span><span class="m">100</span><span class="o">/</span><span class="p">(</span><span class="nf">sum</span><span class="p">(</span><span class="nf">colSums</span><span class="p">(</span><span class="n">data</span><span class="p">)))</span> <span class="o">&gt;</span> <span class="n">percent</span><span class="p">)</span>
    <span class="n">data.filter</span> <span class="o">=</span> <span class="n">data</span><span class="p">[,</span><span class="n">keep.otu</span><span class="p">]</span>
    <span class="nf">return</span><span class="p">(</span><span class="nf">list</span><span class="p">(</span><span class="n">data.filter</span> <span class="o">=</span> <span class="n">data.filter</span><span class="p">,</span> <span class="n">keep.otu</span> <span class="o">=</span> <span class="n">keep.otu</span><span class="p">))</span>
<span class="p">}</span>
</pre></div>
</div>
<p>We run the function on our table, setting up the threshold at 0.02%:</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">result.filter</span> <span class="o">=</span> <span class="nf">low.count.removal</span><span class="p">(</span><span class="nf">t</span><span class="p">(</span><span class="n">table.species.final</span><span class="p">),</span> <span class="n">percent</span><span class="o">=</span><span class="m">0.02</span><span class="p">)</span>
</pre></div>
</div>
<p>We generate a table with the filtered data:</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">table.species.final.flt</span> <span class="o">=</span> <span class="n">result.filter</span><span class="o">$</span><span class="n">data.filter</span>
</pre></div>
</div>
<p>In the next step, we normalize the reads for the sequencing depth. This means that we will account for the total reads generated for each sample, and normalize the species abundance for that number.
To do that we will define a <strong>Total Sum Squared</strong> function</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">TSS.divide</span> <span class="o">=</span> <span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span>
 <span class="n">x</span><span class="o">/</span><span class="nf">sum</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>
<span class="p">}</span>
</pre></div>
</div>
<p>The function is applied to the table, and each row must represent a sample. For this reason we transpose the table.</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">table.species.final.flt.tss</span> <span class="o">=</span> <span class="nf">t</span><span class="p">(</span><span class="nf">apply</span><span class="p">(</span><span class="n">table.species.final.flt</span><span class="p">,</span> <span class="m">1</span><span class="p">,</span> <span class="n">TSS.divide</span><span class="p">))</span>
</pre></div>
</div>
<p>We have just generated a table of species abundances of ancient dental calculus samples, normalized for genome lenghts and sequencing depths.
We can now include in our analysis a dataset of normalized species abundances generated with Kraken2 representing other microbiomes. You can find this table in the server, download it and import it in R.</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">microbiomes.literature</span> <span class="o">=</span> <span class="nf">read.table</span><span class="p">(</span><span class="s">&quot;microbiomes_literature.txt&quot;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="bp">T</span><span class="p">,</span> <span class="n">fill</span><span class="o">=</span><span class="bp">T</span><span class="p">,</span> <span class="n">row.names</span><span class="o">=</span><span class="m">1</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p>The following commands are used to merge the two tables, the dental calculus dataset that you generated and the other microbiomes dataset.</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">table.total</span> <span class="o">=</span> <span class="nf">merge</span><span class="p">(</span><span class="nf">t</span><span class="p">(</span><span class="n">table.species.final.flt.tss</span><span class="p">),</span> <span class="nf">t</span><span class="p">(</span><span class="n">microbiomes.literature</span><span class="p">),</span> <span class="n">by</span><span class="o">=</span><span class="m">0</span><span class="p">,</span> <span class="n">all</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
<span class="n">table.total</span><span class="p">[</span><span class="nf">is.na</span><span class="p">(</span><span class="n">table.total</span><span class="p">)]</span> <span class="o">&lt;-</span> <span class="m">0</span>
<span class="nf">row.names</span><span class="p">(</span><span class="n">table.total</span><span class="p">)</span> <span class="o">=</span> <span class="n">table.total</span><span class="p">[,</span><span class="m">1</span><span class="p">]</span>
<span class="n">table.total.final</span> <span class="o">=</span> <span class="nf">t</span><span class="p">(</span><span class="n">table.total</span><span class="p">[,</span><span class="m">-1</span><span class="p">])</span>
</pre></div>
</div>
<div class="section" id="upgma">
<h4><span class="section-number">7.2.1.1. </span>UPGMA<a class="headerlink" href="#upgma" title="Permalink to this headline">¶</a></h4>
<p>Once generated the final including both datasets (dental calculus and other microbiomes), we run an UPGMA cluster analysis. We must first install the <code class="docutils literal notranslate"><span class="pre">vegan</span></code> and <cite>ape`</cite> package in R.</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">install.packages</span><span class="p">(</span><span class="s">&quot;vegan&quot;</span><span class="p">)</span>
<span class="nf">install.packages</span><span class="p">(</span><span class="s">&quot;ape&quot;</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">vegan</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ape</span><span class="p">)</span>
</pre></div>
</div>
<p>Then we use vegan to calculate the <strong>Bray-Curtis</strong> distances, and run the cluster analysis.</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">bray_dist</span> <span class="o">=</span> <span class="nf">vegdist</span><span class="p">(</span><span class="n">table.total.final</span><span class="p">,</span> <span class="n">method</span> <span class="o">=</span> <span class="s">&quot;bray&quot;</span><span class="p">,</span> <span class="n">binary</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">diag</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">upper</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> <span class="n">na.rm</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
<span class="n">bray_dist.clust</span> <span class="o">=</span> <span class="nf">hclust</span><span class="p">(</span><span class="n">bray_dist</span><span class="p">,</span> <span class="n">method</span><span class="o">=</span><span class="s">&quot;average&quot;</span><span class="p">,</span> <span class="n">members</span> <span class="o">=</span> <span class="kc">NULL</span><span class="p">)</span>
</pre></div>
</div>
<p>Finally, we plot the dendrogram:</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">plot</span><span class="p">(</span><span class="nf">as.phylo</span><span class="p">(</span><span class="n">bray_dist.clust</span><span class="p">),</span> <span class="n">type</span> <span class="o">=</span> <span class="s">&quot;unrooted&quot;</span><span class="p">,</span> <span class="n">cex</span> <span class="o">=</span> <span class="m">0.5</span><span class="p">,</span> <span class="n">lab4ut</span><span class="o">=</span><span class="s">&quot;axial&quot;</span><span class="p">,</span> <span class="n">no.margin</span><span class="o">=</span><span class="bp">T</span><span class="p">,</span> <span class="n">show.tip.label</span><span class="o">=</span><span class="bp">T</span><span class="p">,</span> <span class="n">label.offset</span><span class="o">=</span><span class="m">0.02</span><span class="p">,</span> <span class="n">edge.color</span> <span class="o">=</span> <span class="s">&quot;gray&quot;</span><span class="p">,</span> <span class="n">edge.width</span> <span class="o">=</span> <span class="m">1</span><span class="p">,</span> <span class="n">edge.lty</span> <span class="o">=</span> <span class="m">1</span><span class="p">)</span>
</pre></div>
</div>
<p>To visualize better our samples, we can define colors. We will assign group labels on each sample:</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">labels</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;Velsko-ancient&quot;</span><span class="p">,</span><span class="s">&quot;Velsko-ancient&quot;</span><span class="p">,</span><span class="s">&quot;Velsko-ancient&quot;</span><span class="p">,</span><span class="s">&quot;Velsko-ancient&quot;</span><span class="p">,</span><span class="s">&quot;Velsko-ancient&quot;</span><span class="p">,</span><span class="s">&quot;Velsko-ancient&quot;</span><span class="p">,</span><span class="s">&quot;Velsko-ancient&quot;</span><span class="p">,</span><span class="s">&quot;Velsko-ancient&quot;</span><span class="p">,</span><span class="s">&quot;Velsko-ancient&quot;</span><span class="p">,</span><span class="s">&quot;Velsko-ancient&quot;</span><span class="p">,</span>
                                              <span class="s">&quot;Velsko-modern&quot;</span><span class="p">,</span><span class="s">&quot;Velsko-modern&quot;</span><span class="p">,</span><span class="s">&quot;Velsko-modern&quot;</span><span class="p">,</span><span class="s">&quot;Velsko-modern&quot;</span><span class="p">,</span><span class="s">&quot;Velsko-modern&quot;</span><span class="p">,</span><span class="s">&quot;Velsko-modern&quot;</span><span class="p">,</span><span class="s">&quot;Velsko-modern&quot;</span><span class="p">,</span><span class="s">&quot;Velsko-modern&quot;</span><span class="p">,</span>
                                              <span class="s">&quot;Ancient calculus&quot;</span><span class="p">,</span><span class="s">&quot;Ancient tooth&quot;</span><span class="p">,</span><span class="s">&quot;Ancient calculus&quot;</span><span class="p">,</span><span class="s">&quot;Ancient tooth&quot;</span><span class="p">,</span>
                                              <span class="s">&quot;Soil&quot;</span><span class="p">,</span><span class="s">&quot;Soil&quot;</span><span class="p">,</span><span class="s">&quot;Soil&quot;</span><span class="p">,</span><span class="s">&quot;Soil&quot;</span><span class="p">,</span><span class="s">&quot;Soil&quot;</span><span class="p">,</span><span class="s">&quot;Soil&quot;</span><span class="p">,</span><span class="s">&quot;Soil&quot;</span><span class="p">,</span>
                                              <span class="s">&quot;Ancient calculus&quot;</span><span class="p">,</span><span class="s">&quot;Ancient tooth&quot;</span><span class="p">,</span><span class="s">&quot;Ancient calculus&quot;</span><span class="p">,</span><span class="s">&quot;Ancient tooth&quot;</span><span class="p">,</span><span class="s">&quot;Ancient calculus&quot;</span><span class="p">,</span><span class="s">&quot;Ancient tooth&quot;</span><span class="p">,</span><span class="s">&quot;Ancient calculus&quot;</span><span class="p">,</span><span class="s">&quot;Ancient tooth&quot;</span><span class="p">,</span><span class="s">&quot;Ancient calculus&quot;</span><span class="p">,</span><span class="s">&quot;Ancient tooth&quot;</span><span class="p">,</span><span class="s">&quot;Ancient calculus&quot;</span><span class="p">,</span><span class="s">&quot;Ancient tooth&quot;</span><span class="p">,</span>
                                              <span class="s">&quot;Plaque&quot;</span><span class="p">,</span><span class="s">&quot;Plaque&quot;</span><span class="p">,</span><span class="s">&quot;Plaque&quot;</span><span class="p">,</span><span class="s">&quot;Plaque&quot;</span><span class="p">,</span><span class="s">&quot;Plaque&quot;</span><span class="p">,</span><span class="s">&quot;Plaque&quot;</span><span class="p">,</span><span class="s">&quot;Plaque&quot;</span><span class="p">,</span><span class="s">&quot;Plaque&quot;</span><span class="p">,</span><span class="s">&quot;Plaque&quot;</span><span class="p">,</span><span class="s">&quot;Plaque&quot;</span><span class="p">,</span>
                                              <span class="s">&quot;Plaque&quot;</span><span class="p">,</span><span class="s">&quot;Plaque&quot;</span><span class="p">,</span><span class="s">&quot;Plaque&quot;</span><span class="p">,</span><span class="s">&quot;Plaque&quot;</span><span class="p">,</span><span class="s">&quot;Plaque&quot;</span><span class="p">,</span><span class="s">&quot;Plaque&quot;</span><span class="p">,</span><span class="s">&quot;Plaque&quot;</span><span class="p">,</span><span class="s">&quot;Plaque&quot;</span><span class="p">,</span><span class="s">&quot;Plaque&quot;</span><span class="p">,</span><span class="s">&quot;Plaque&quot;</span><span class="p">,</span>
                                              <span class="s">&quot;Plaque&quot;</span><span class="p">,</span><span class="s">&quot;Plaque&quot;</span><span class="p">,</span><span class="s">&quot;Plaque&quot;</span><span class="p">,</span><span class="s">&quot;Plaque&quot;</span><span class="p">,</span><span class="s">&quot;Plaque&quot;</span><span class="p">,</span>
                                              <span class="s">&quot;Skin&quot;</span><span class="p">,</span><span class="s">&quot;Skin&quot;</span><span class="p">,</span><span class="s">&quot;Skin&quot;</span><span class="p">,</span><span class="s">&quot;Skin&quot;</span><span class="p">,</span>
                                              <span class="s">&quot;Gut&quot;</span><span class="p">,</span><span class="s">&quot;Gut&quot;</span><span class="p">,</span><span class="s">&quot;Gut&quot;</span><span class="p">,</span><span class="s">&quot;Gut&quot;</span><span class="p">,</span><span class="s">&quot;Gut&quot;</span><span class="p">,</span><span class="s">&quot;Gut&quot;</span><span class="p">,</span><span class="s">&quot;Gut&quot;</span><span class="p">,</span><span class="s">&quot;Gut&quot;</span><span class="p">,</span><span class="s">&quot;Gut&quot;</span><span class="p">,</span><span class="s">&quot;Gut&quot;</span><span class="p">,</span>
                                              <span class="s">&quot;Gut&quot;</span><span class="p">,</span><span class="s">&quot;Gut&quot;</span><span class="p">,</span><span class="s">&quot;Gut&quot;</span><span class="p">,</span><span class="s">&quot;Gut&quot;</span><span class="p">,</span><span class="s">&quot;Gut&quot;</span><span class="p">,</span><span class="s">&quot;Gut&quot;</span><span class="p">,</span><span class="s">&quot;Gut&quot;</span><span class="p">,</span><span class="s">&quot;Gut&quot;</span><span class="p">,</span><span class="s">&quot;Gut&quot;</span><span class="p">,</span><span class="s">&quot;Gut&quot;</span><span class="p">,</span>
                                              <span class="s">&quot;Gut&quot;</span><span class="p">,</span><span class="s">&quot;Gut&quot;</span><span class="p">,</span><span class="s">&quot;Gut&quot;</span><span class="p">,</span><span class="s">&quot;Gut&quot;</span><span class="p">,</span>
                                              <span class="s">&quot;Skin&quot;</span><span class="p">,</span><span class="s">&quot;Skin&quot;</span><span class="p">,</span><span class="s">&quot;Skin&quot;</span><span class="p">,</span><span class="s">&quot;Skin&quot;</span><span class="p">,</span><span class="s">&quot;Skin&quot;</span><span class="p">,</span>
                                              <span class="s">&quot;Plaque&quot;</span><span class="p">,</span><span class="s">&quot;Plaque&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p>To have a better look at the correspondence of data we can create a dataframe:</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">table.total.final.df</span> <span class="o">=</span> <span class="nf">as.data.frame</span><span class="p">(</span><span class="n">table.total.final</span><span class="p">)</span>
<span class="n">table.total.final.df</span><span class="o">$</span><span class="n">group</span> <span class="o">=</span> <span class="n">labels</span>
</pre></div>
</div>
<p>We assign colors to each label:</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">coul</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;#E41A1C&quot;</span><span class="p">,</span>             <span class="c1">#Ancient calculus</span>
              <span class="s">&quot;#419681&quot;</span><span class="p">,</span>              <span class="c1">#Ancient tooth</span>
              <span class="s">&quot;#4DAF4A&quot;</span><span class="p">,</span>              <span class="c1">#Gut</span>
              <span class="s">&quot;lightgray&quot;</span><span class="p">,</span>    <span class="c1">#Plaque</span>
              <span class="s">&quot;#984EA3&quot;</span><span class="p">,</span>              <span class="c1">#Skin</span>
              <span class="s">&quot;#FF7F00&quot;</span><span class="p">,</span>              <span class="c1">#Soil</span>
              <span class="s">&quot;goldenrod&quot;</span><span class="p">,</span>    <span class="c1">#Velsko-ancient</span>
              <span class="s">&quot;#994C00&quot;</span><span class="p">)</span>              <span class="c1">#Velsko-modern</span>
</pre></div>
</div>
<p>And finally, we plot again the dendrogram, this time by customizing the tips assigning color-coded labels:</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">plot</span><span class="p">(</span><span class="nf">as.phylo</span><span class="p">(</span><span class="n">bray_dist.clust</span><span class="p">),</span> <span class="n">type</span> <span class="o">=</span> <span class="s">&quot;unrooted&quot;</span><span class="p">,</span> <span class="n">cex</span> <span class="o">=</span> <span class="m">0.5</span><span class="p">,</span> <span class="n">lab4ut</span><span class="o">=</span><span class="s">&quot;axial&quot;</span><span class="p">,</span> <span class="n">no.margin</span><span class="o">=</span><span class="bp">T</span><span class="p">,</span> <span class="n">show.tip.label</span><span class="o">=</span><span class="bp">T</span><span class="p">,</span> <span class="n">label.offset</span><span class="o">=</span><span class="m">0.02</span><span class="p">,</span> <span class="n">edge.color</span> <span class="o">=</span> <span class="s">&quot;gray&quot;</span><span class="p">,</span> <span class="n">edge.width</span> <span class="o">=</span> <span class="m">1</span><span class="p">,</span> <span class="n">edge.lty</span> <span class="o">=</span> <span class="m">1</span><span class="p">)</span>
<span class="nf">tiplabels</span><span class="p">(</span><span class="n">pch</span><span class="o">=</span><span class="m">19</span><span class="p">,</span> <span class="n">col</span> <span class="o">=</span> <span class="n">coul</span><span class="p">[</span><span class="nf">factor</span><span class="p">(</span><span class="n">labels</span><span class="p">)],</span> <span class="n">bg</span> <span class="o">=</span> <span class="n">coul</span><span class="p">[</span><span class="nf">factor</span><span class="p">(</span><span class="n">labels</span><span class="p">)],</span> <span class="n">cex</span><span class="o">=</span><span class="m">1</span><span class="p">,</span> <span class="n">lwd</span><span class="o">=</span><span class="m">1</span><span class="p">)</span>
</pre></div>
</div>
<p>And we can add a legend:</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">legend</span><span class="p">(</span><span class="s">&quot;topleft&quot;</span><span class="p">,</span> <span class="n">legend</span> <span class="o">=</span> <span class="nf">sort</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">labels</span><span class="p">)),</span> <span class="n">bty</span> <span class="o">=</span> <span class="s">&quot;n&quot;</span><span class="p">,</span> <span class="n">col</span> <span class="o">=</span> <span class="n">coul</span><span class="p">,</span> <span class="n">pch</span> <span class="o">=</span> <span class="m">19</span><span class="p">,</span> <span class="n">pt.cex</span><span class="o">=</span><span class="m">1</span><span class="p">,</span> <span class="n">cex</span><span class="o">=</span><span class="m">0.6</span><span class="p">,</span> <span class="n">pt.lwd</span><span class="o">=</span><span class="m">1</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="10_old_material.html" class="btn btn-neutral float-right" title="8. Material from previous courses" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="8_Filtering_SNPs.html" class="btn btn-neutral float-left" title="6. Filtering, annotating and combining SNPs" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, Amine Namouchi &amp; Claudio Ottoni

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
    <!-- Theme Analytics -->
    <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-XXXXXXX-1', 'auto');
    ga('send', 'pageview');
    </script>

    
   

</body>
</html>